newdle/client/src/components/answer/AnswerPage.js

```diff
@@ -12,14 +15,34 @@ import {
 } from '../../answerSelectors';
 import {chooseAllAvailable} from '../../actions';
 import styles from './answer.module.scss';
+import client from '../../client';
 
 export default function AnswerPage() {
+  const {partcode: participantCode} = useParams();
   const dispatch = useDispatch();
   const newdle = useSelector(getNewdle);
   const numberOfTimeslots = useSelector(getNumberOfTimeslots);
   const numberOfAvailable = useSelector(getNumberOfAvailableAnswers);
+  const availabilityData = useSelector(getAnswers);
   const allAvailableSelected = useSelector(isAllAvailableSelected);
   const allAvailableDisabled = useSelector(isAllAvailableSelectedImplicitly);
+  const [name, setName] = useState('');
+  const history = useHistory();
+  const [saved, setSaved] = useState(false);
+
+  const [submitAnswer, submitting, error] = participantCode
+    ? client.useBackend(client.updateParticipantAnswers)
+    : client.useBackend(client.createAnonymousParticipant, ({code}) =>
+        client.updateParticipantAnswers(newdle.code, code, availabilityData)
+      );
+
+  const canSubmit = (participantCode || name.length >= 2) && !submitting;
+
+  async function answerNewdle() {
+    const {code} = await submitAnswer(newdle.code, participantCode || name, availabilityData);
+    history.push(`/newdle/${newdle.code}/${code}`);
```

Comment:
You can end up in an inconsistent state if the answer request fails but participant creation worked. Should we push the history change right after participant instead?

Also, this should only run if a new participant was created, or you get an extra history entry you don't need.

Reply:
Yeah, about that second part, I had added it but then lost the change during one of the rebases.